packages/dart/sshnoports/lib/src/noports_cli/issue_keys/np_issue_keys.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:sshnoports/src/noports_cli/util/np_utils.dart';
+
+sealed class NPIssueKeys {
+  /// Entry point for the issue-keys command
+  Future<int> wrappedMain(List<String> args);
+
+  static final baseEnrollCommand = '<atsign>:enroll:otp:<otp>';
+  static final defaultDeviceNamePrefix = 'noports_';
+}
+
+class NPIssueKeysImpl implements NPIssueKeys {
+  final AtServiceFactory _factory = DefaultAtServiceFactory();
+  EnrollmentService? _enrollmentService;
+
+  @override
+  Future<int> wrappedMain(List<String> args) async {
+    NoportsParams params = NoportsParams.fromArgs(args);
+    // fetch atKeys filePath from user (can be null)
+    params.atKeysFilePath = _promptUser('atKeys filepath (target location): ');
+    writeWarning('atkeys: \'${params.atKeysFilePath}\'');
+    final atClient = await createAtClientCli(
+        atsign: params.atsign,
+        atServiceFactory: _factory,
+        namespace: defaultCurrentNamespace,
+        storagePath: standardAtClientStoragePath(
+            baseDir: getHomeDirectory()!,
+            atSign: params.atsign,
+            progName: defaultCurrentNamespace));
+
+    _enrollmentService = DefaultAtServiceFactory().enrollmentService(atClient);
+
+    // fetch enrollment OTP using at_client instance
+    params.otp = await requestEnrollmentOtp(atClient);
+
+    //fetch deviceName from user (defaults to: noports_<otp>)
+    params.deviceName = _promptUser('deviceName: ') ??
+        '${NPIssueKeys.defaultDeviceNamePrefix}${params.otp}';
+
+    String command = _generateEnrollCommand(params);
+    writeInfoMessage('Copy the string below \t\'noports activate <string>\'\n'
+        '\n\tFor Noports Desktop use:\n\t$command\n'
+        '\n\tFor Noports CLI use:\n\tnoports activate \'$command\'\n');
+
+    AtEnrollmentResponse er = await _approveFirstPendingEnrollment(params);
+    if (er.enrollStatus != EnrollmentStatus.approved) {
+      writeError('Failed to approve enrollment\nStatus: $er');
+      return 1;
+    }
+    writeSuccessMessage('Enrollment approved: ${er.enrollmentId}\n');
+    return 0;
+  }
+
+  String? _promptUser(String prompt) {
+    stdout.write(prompt);
+    String? input = stdin.readLineSync();
+    if (input == null) writeWarning('Missing input, using default');
+    // ensure output is either user input or null
+    return (input == null || input.isEmpty) ? null : input;
+  }
+
+  /// Generates an activation string from the provided [params]
+  ///
+  /// Combines atsign, otp, and optional deviceName and atKeysFilePath
+  /// into a format compatible with 'noports activate' command.
+  ///
+  /// Returns: activation string in format `<atsign>:enroll:otp:<otp>[:name:<deviceName>[:keyfile:<path>]]`
+  String _generateEnrollCommand(NoportsParams params) {
+    StringBuffer cbuf = StringBuffer();
+    cbuf.append(NPIssueKeys.baseEnrollCommand
+        .replaceFirst('<atsign>', params.atsign)
+        .replaceFirst('<otp>', params.otp!));
```

Comment:
baseEnrollCommand is `<atsign>:enroll:otp:<otp>`; replace placeholders with actual values